File: codes/questao02/fibonacci.c
#include <stdio.h>

int isFibonacci(int num)
{
    int a = 0, b = 1, c;

    while (c <= num)
    {
        if (c == num)
        {
            return 1;
        }
        c = a + b;
        a = b;
        b = c;
    }

    return 0;
}

int main()
{
    int numero;

    printf("Informe um número: ");
    scanf("%d", &numero);

    if (isFibonacci(numero))
    {
        printf("%d pertence à sequência de Fibonacci.\n", numero);
    }
    else
    {
        printf("%d não pertence à sequência de Fibonacci.\n", numero);
    }

    return 0;
}

File: codes/questao02/fibonacci.js
function isFibonacci(num) {
  let a = 0,
    b = 1,
    c;

  while (c <= num) {
    if (c === num) {
      return true;
    }
    c = a + b;
    a = b;
    b = c;
  }

  return false;
}

let numero = parseInt(prompt("Informe um número:"));

if (isFibonacci(numero)) {
  console.log(`${numero} pertence à sequência de Fibonacci.`);
} else {
  console.log(`${numero} não pertence à sequência de Fibonacci.`);
}
